### ISSUE TITLE
"Dotlottie works fine on localhost but not in production" - content type detection causes animation not to load.

### ISSUE THREAD

**Issue by invernizzi** [LottieFiles/dotlottie-web#282]
Hello there!

I want to share a DotLottie quirk, in the hope it can save someone else time.

Specifically: DotLottie decides how to load a particular animation (in the lottie or dotlottie format) based on the **content type**
of the URL containing the animation data. 

These are the [lines](https://github.com/LottieFiles/dotlottie-web/blob/8ca8fd9947ef7902cb0198009c79e56808dbfe52/packages/web/src/dotlottie.ts#L142) making that decision.
```
      if (['application/json', 'text/plain'].some((type) => contentType.startsWith(type))) {
        data = await response.text();
      } else {
        data = await response.arrayBuffer();
      }
```

This behavior causes a tricky lite quirk with Vite.

In development mode, files imported like so are served with an **empty** content type. That causes dotlottie to work correctly if that file is a dotlottie one (`arrayBuffer()` is called).
```
import walkAnimationDataUrl from "@/public/walk.lottie?url";
```

However! In production, the same line is served with a content type of `text/html`. Dotlottie will load it as text (`text()` is called), but it won't raise any error (at least, with my file). Instead, it will cause the browser process to hang after a few seconds.

DotLottie devs: perhaps you could let users specify the file format as an option instead of this. Since I'm using Vite with pretty much default options, I wonder if I'm the first to face this issue.

(I need a celebratory beer after solving this one)

**Comment by invernizzi** [comment]
Thanks for the quick response!

your reasoning makes sense, but that's a dotlottie file, not a lottie one. I'm not sure why the use the same extension by convension (as opposed to `.dotlottie`), but that's the convention followed by the [site describing the file format](https://dotlottie.io/structure/#interactivity-structure). 

I've created a simple reproduction of this bug in this repo, using an example animation from [that same site](https://dotlottie.io/structure/#interactivity-structure). Here it is: https://github.com/invernizzi/dotlottie-web-bug-282

**Comment by theashraf** [comment]
@invernizzi thanks for reproducing the issue. It seems that Vite can't recognize the .lottie file type and add the proper content-type header accordingly.

On preview:
![image](https://github.com/LottieFiles/dotlottie-web/assets/39750790/2baaec07-08de-431d-9613-323073bc0d0a)

On dev:
![image](https://github.com/LottieFiles/dotlottie-web/assets/39750790/84277a73-5489-44da-bc96-0a8100814653)

In my opinion, one way we can improve this from the dotlottie-web side is to resolve all src's as arrayBuffers and check for the ZIP magic number. Given the fact that .lottie is just a ZIP archive, we can use the signature of a ZIP file to identify it. I believe the ZIP signature in hex is: 50 4B 03 04

I feel that this is a good approach, as it is a common practice used to recognize file types. If this signature is not present, then it's a Lottie JSON file.

**Comment by x5hwuk7nc6jsudcc** [comment]
👏 great issue and solution - can we get this merged in to a release?

**Comment by theashraf** [comment]
@x5hwuk7nc6jsudcc I've raised the PR for review, and it will be merged soon

**Comment by theashraf** [comment]
has been released in dotlottie-web@v0.29.2
